src/runtime/rpc/rpc_host_port.h

```diff
@@ -118,6 +130,22 @@ inline bool operator==(const host_port &hp1, const host_port &hp2)
 
 inline bool operator!=(const host_port &hp1, const host_port &hp2) { return !(hp1 == hp2); }
 
+inline bool operator<(const host_port &hp1, const host_port &hp2)
+{
+    if (hp1.type() != hp2.type()) {
+        return hp1.type() < hp2.type();
+    }
+
+    switch (hp1.type()) {
+    case HOST_TYPE_IPV4:
+        return hp1.host() < hp2.host() || (hp1.host() == hp2.host() && hp1.port() < hp2.port());
+    case HOST_TYPE_GROUP:
+        return reinterpret_cast<uint64_t>(hp1.group_host_port().get()) <
+               reinterpret_cast<uint64_t>(hp2.group_host_port().get());
```

Comment:
Can be replaced by this?
```suggestion
        return hp1.group_host_port().get() < hp2.group_host_port().get();
```

Reply:
I think it's better to be specific.